ReactAndroid/src/main/java/com/facebook/react/ReactFragmentActivity.java

```diff
@@ -7,126 +7,11 @@
-import com.facebook.react.modules.core.PermissionListener;
-
 /**
  * @deprecated
  * ReactFragmentActivity will be removed in 0.57 release.
  * Use {@link ReactActivity} instead.
  */
 @Deprecated
-public abstract class ReactFragmentActivity extends FragmentActivity implements
-  DefaultHardwareBackBtnHandler, PermissionAwareActivity {
-
-  private final ReactActivityDelegate mDelegate;
-
-  protected ReactFragmentActivity() {
-    mDelegate = createReactActivityDelegate();
-  }
-
-  /**
-   * Returns the name of the main component registered from JavaScript.
-   * This is used to schedule rendering of the component.
-   * e.g. "MoviesApp"
-   */
-  protected @Nullable String getMainComponentName() {
-    return null;
-  }
-
-  /**
-   * Called at construction time, override if you have a custom delegate implementation.
-   */
-  protected ReactActivityDelegate createReactActivityDelegate() {
-    return new ReactActivityDelegate(this, getMainComponentName());
-  }
-
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    mDelegate.onCreate(savedInstanceState);
-  }
-
-  @Override
-  protected void onPause() {
-    super.onPause();
-    mDelegate.onPause();
-  }
-
-  @Override
-  protected void onResume() {
-    super.onResume();
-    mDelegate.onResume();
-  }
-
-  @Override
-  protected void onDestroy() {
-    super.onDestroy();
-    mDelegate.onDestroy();
-  }
-
-  @Override
-  public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    mDelegate.onActivityResult(requestCode, resultCode, data);
-  }
-
-  @Override
-  public boolean onKeyUp(int keyCode, KeyEvent event) {
-    return mDelegate.onKeyUp(keyCode, event) || super.onKeyUp(keyCode, event);
-  }
-
-  @Override
-  public void onBackPressed() {
-    if (!mDelegate.onBackPressed()) {
-      super.onBackPressed();
-    }
-  }
-
-  @Override
-  public void invokeDefaultOnBackPressed() {
-    super.onBackPressed();
-  }
-
-  @Override
-  public void onNewIntent(Intent intent) {
-    if (!mDelegate.onNewIntent(intent)) {
-      super.onNewIntent(intent);
-    }
-  }
-
-  @Override
-  public void requestPermissions(
-    String[] permissions,
-    int requestCode,
-    PermissionListener listener) {
-    mDelegate.requestPermissions(permissions, requestCode, listener);
-  }
-
-  @Override
-  public void onRequestPermissionsResult(
-    int requestCode,
-    String[] permissions,
-    int[] grantResults) {
-    mDelegate.onRequestPermissionsResult(requestCode, permissions, grantResults);
-  }
-
-  protected final ReactNativeHost getReactNativeHost() {
-    return mDelegate.getReactNativeHost();
-  }
-
-  protected final ReactInstanceManager getReactInstanceManager() {
-    return mDelegate.getReactInstanceManager();
-  }
-
-  protected final void loadApp(String appKey) {
-    mDelegate.loadApp(appKey);
-  }
+public abstract class ReactFragmentActivity extends ReactActivity {
```

Comment:
What's the point of keeping this activity with no subclassed methods of ReactActivity? Looks like it could be removed now.

Reply:
I think that the intention is to maintain backwards compatibility. Some existing projects currently extends from `ReactFragmentActivity`.